pipelines/pipeline.yml

```diff
@@ -76,23 +70,17 @@ stages:
       - job: Build
         dependsOn: Test
         steps:
+          # Build go executables
           - script: |
               make build
             displayName: "Build"
+
+          # Publish build artifacts
+          # for deploying to PR environment, and for Github releases
           - publish: $(System.DefaultWorkingDirectory)/bin
             artifact: bin
-          - publish: $(System.DefaultWorkingDirectory)/modules
-            artifact: modules
           - publish: $(System.DefaultWorkingDirectory)/scripts
             artifact: deploy_scripts
-          - publish: $(System.DefaultWorkingDirectory)/tests
```

Comment:
I mucked around a bunch with how files are shared between jobs.
I added `checkout: true` to all the stages, so everything has access to the main repo code,
and we only need to move generated artifacts around.

Not necessarily better or worse this way, but I was having a rough time of it, 
and it's working now!

Reply:
Tried to find that type of property but was having really bad luck.  Nice find.

Reply:
yeah, and it's weird because _some_ stages do it automatically, and some don't.